crates/biome_js_analyze/src/analyzers/nursery/no_useless_ternary.rs

```diff
@@ -97,3 +204,66 @@ fn is_boolean_literal(expression: &AnyJsExpression) -> bool {
     }
     false
 }
+
+fn is_boolean_expression(expression: &AnyJsExpression) -> Option<bool> {
+    match expression.syntax().kind() {
+        JsSyntaxKind::JS_BINARY_EXPRESSION => {
+            let operator = expression
+                .as_js_binary_expression()?
+                .operator_token()
+                .ok()?;
+            let operator = operator.text_trimmed();
+            if BOOLEAN_OPERATORS.contains(&operator) {
+                return Some(true);
+            }
```

Comment:
We have a bunch of utilities for operators; use those :) 

https://github.com/biomejs/biome/blob/main/crates/biome_js_syntax/src/expr_ext.rs#L182

Also, `instanceof` and `in` have different AST, so you will never detect them here: https://biomejs.dev/playground/?code=ZgAgAGkAbgBzAHQAYQBuAGMAZQBvAGYAIABDAGwAYQBzAHMA

In fact, I don't see any tests around those two.